pkg/console/controllers/poddisruptionbudget/contrroller.go

```diff
@@ -0,0 +1,119 @@
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	// console-operator
+	"github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/console/controllers/util"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+type PodDisruptionBudgetController struct {
+	pdbName              string
+	operatorClient       v1helpers.OperatorClient
+	operatorConfigClient operatorclientv1.ConsoleInterface
+	pdbClient            policyv1client.PodDisruptionBudgetsGetter
+}
+
+func NewPodDisruptionBudgetController(
+	// name of the PDB instance
+	pdbName string,
+
+	// clients
+	operatorClient v1helpers.OperatorClient,
+	operatorConfigClient operatorclientv1.ConsoleInterface,
+	pdbClient policyv1client.PodDisruptionBudgetsGetter,
+
+	// informer
+	pdbInformer policyv1.PodDisruptionBudgetInformer,
+
+	//events
+	recorder events.Recorder,
+) factory.Controller {
+
+	ctrl := &PodDisruptionBudgetController{
+		operatorClient:       operatorClient,
+		operatorConfigClient: operatorConfigClient,
+	}
+
+	return factory.New().
+		WithFilteredEventsInformers(
+			util.IncludeNamesFilter(pdbName),
+			pdbInformer.Informer(),
+		).ResyncEvery(time.Minute).WithSync(ctrl.Sync).
+		ToController("PodDisruptionBudgetController", recorder.WithComponentSuffix(fmt.Sprintf("%s-pdb-controller", pdbName)))
+}
+
+func (c *PodDisruptionBudgetController) Sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	operatorConfig, err := c.operatorConfigClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	updatedOperatorConfig := operatorConfig.DeepCopy()
+
+	switch updatedOperatorConfig.Spec.ManagementState {
+	case operatorsv1.Managed:
+		klog.V(4).Infof("console-operator is in a managed state: syncing %q pdb", c.pdbName)
+	case operatorsv1.Unmanaged:
+		klog.V(4).Infof("console-operator is in an unmanaged state: skipping pdb %q sync", c.pdbName)
+		return nil
+	case operatorsv1.Removed:
+		klog.V(4).Infof("console-operator is in a removed state: deleting %q pdb", c.pdbName)
+		if err = c.removePodDisruptionBudget(ctx, c.pdbName); err != nil {
+			return err
+		}
+		return c.removePodDisruptionBudget(ctx, c.pdbName)
+	default:
+		return fmt.Errorf("unknown state: %v", updatedOperatorConfig.Spec.ManagementState)
+	}
+
+	statusHandler := status.NewStatusHandler(c.operatorClient)
+
+	requiredPDB := c.getDefaultPodDisruptionBudget()
+	_, _, pdbErr := resourceapply.ApplyPodDisruptionBudget(ctx, c.pdbClient, controllerContext.Recorder(), requiredPDB)
+	statusHandler.AddConditions(status.HandleProgressingOrDegraded("PDBSync", "FailedApply", pdbErr))
+	if pdbErr != nil {
+		return statusHandler.FlushAndReturn(pdbErr)
+	}
+
+	return statusHandler.FlushAndReturn(pdbErr)
+
+}
+
+// Remove the PDB instance the controller is managing
+func (c *PodDisruptionBudgetController) removePodDisruptionBudget(ctx context.Context, pdbName string) error {
```

Comment:
no need to pass the name, you can use `c.pdbName` from the instance
```suggestion
func (c *PodDisruptionBudgetController) removePodDisruptionBudget(ctx context.Context) error {
```